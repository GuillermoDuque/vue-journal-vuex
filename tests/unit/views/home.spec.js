import { Exception } from 'sass'
import Home from '@/views/HomeView'
import { shallowMount } from "@vue/test-utils"

describe('Pruebas en el Home View',()=>{
    test('debe de renderizar el componente correctamente',() =>{
        
        const wrapper = shallowMount(Home)
        expect( wrapper.html() ).toMatchSnapshot()
        
    })

    test('hacer click en un botÃ³n debe de redireccionar a no-entry',() =>{
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router:mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect( mockRouter.push).toHaveBeenCalled()
        expect( mockRouter.push).toHaveBeenCalledWith({name:'no-entry'})
    })
})