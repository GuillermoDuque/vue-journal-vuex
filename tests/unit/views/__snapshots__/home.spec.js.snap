// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home View debe de renderizar el componente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <helloworld msg="Welcome to Your Vue.js App"></helloworld><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Primary</button><button class="btn btn-success">Primary</button>
</div>
`;
